### PR TITLE
Release 0.10.0 

### LINKED ISSUE #20: Copy/Move fast paths not used when casting to interface
I noticed that for the interface check using the generic `T`:
https://github.com/Arlodotexe/OwlCore.Storage/blob/8baab6c92a9dd7195ffc8ad0b098d0fd1879c929/src/Extensions/CopyAndMoveExtensions.cs#L25

When `T` is declared as an interface instead of an implementation, this type check always fails and the fastpath for the specific `T fileToCopy` implementation is not selected. 

Unless you explicitly provide the type for the file to copy or move (not the interface), you'll always get the slower fallback code path. That breaks fastpaths unless you explicitly provide the type you meant, and that is slowing down some inbox tooling by avoiding fastpaths.

Might need to expose the generic parameter in more places if we want generic fastpaths to work 100% of the time. 

This only affects copy/move operations for now and will still work as expected, but it's still something for consumers to keep in mind in the meantime.

### LINKED ISSUE #28: Better parameter validation for SystemFile ctor
**Issue**
Exception raised is a FileNotFoundException, this is incorrect. 

**Reason**
The exception will be raised for many different reasons, for example, an invalid path, null, empty, or the user does not have permissions

**Proposal**
Either fire the correct exception, remove exceptions (issue #24), or remove the check in its entirety and use properties (see issue #27)

### LINKED ISSUE #36: Some async methods are missing cancellation tokens
These extension methods and interfaces are missing a `CancellationToken` as a parameter:

https://github.com/Arlodotexe/OwlCore.Storage/blob/75312821cb0a25938a0fcefeaaa2a0c2f033a4c4/src/Extensions/GetRelativePathToExtension.cs#L13
https://github.com/Arlodotexe/OwlCore.Storage/blob/75312821cb0a25938a0fcefeaaa2a0c2f033a4c4/src/Extensions/GetRootExtension.cs#L15

https://github.com/Arlodotexe/OwlCore.Storage/blob/75312821cb0a25938a0fcefeaaa2a0c2f033a4c4/src/Extensions/IFastGetRoot.cs#L8

### LINKED ISSUE #46: Rename IFast.. interfaces
## Motivation
Currently, there are so-called "fast-path" interfaces that omit base operations, if an implementation supports a more efficient alternative. The prefix "Fast" is not effective in this measure since the operation can still take more time than perceived "fast".

```cs
public interface IFastFileCopy<in T> : IModifiableFolder where T : IFile
```
```cs
public interface IFastGetFirstByName : IFolder
```
copying or enumerating large collections is not fast by any means.

## Proposal
I propose to rename all interfaces prefixed with "Fast" to "Direct" as it better reflects the intentions of such interfaces. They are meant to provide an alternative way to **directly** access a certain functionality, not necessarily fast.

```cs
public interface IDirectFileCopy<in T> : IModifiableFolder where T : IFile
```
```cs
public interface IDirectGetFirstByName : IFolder
```

### LINKED ISSUE #48: Align namespaces with .NET where possible
Posting from [this discussion](https://discord.com/channels/372137812037730304/1211662592192286720) in the UWP Community Discord. 

We don't know how much .NET will add in the future that can be implemented "inbox", so we should match the .NET namespaces in our own. For example:
- `OwlCore.Storage.System.IO`
- `OwlCore.Storage.System.IO.Compression`
- `OwlCore.Storage.System.Formats.Tar`
- `OwlCore.Storage.System.Net.Http`

This is a large breaking change and is planned for v0.10.0.

### PR DESCRIPTION
**Pull request by Arlodotexe**
This PR contains the remaining changes needed to release the 0.10.0 update.

- See also https://github.com/Arlodotexe/OwlCore.Storage/pull/49
- 5e0a298e2246ef8adb7522d69b0b0c51e878d90f fixes #20
- f23ba0c6c9e56e63e855711ceb6388750d126474 fixes #36
- 11287bc8a3292b42f2ed8671746fe44ce731fbc8 fixes #46
- c35046e3099615ad8b152b1c3b97118e0c5466c2 fixes #28
- e3e1a4b781bfe6ef14d4ab34a3d04efed6700fbf fixes #48

Special thanks to all the devs who provided their honest feedback, both on GitHub and over in the UWP Community Discord.
We've come a long way since the initial proposal (almost 2 years!) and we're very nearly ready to submit our [CommunityToolkit.Storage](https://github.com/CommunityToolkit/Labs-Windows/discussions/229) PR to Toolkit Labs. Get hyped!

### CODE REVIEW

**Comment by Arlodotexe** [comment]
Draft PR while I work on the other tickets for 0.10.0. I'll be doing them all in one sweep using the `dev/0.10.0` branch, and closing it in this PR.

**Comment by Arlodotexe** [comment]
Leaving this open 24hrs for community review. Please note that tests in CI will be broken until the 0.5.0 update for our [CommonTests](https://github.com/Arlodotexe/OwlCore.Storage.CommonTests) can go out with 0.10.0 support.

File: src/System/IO/SystemFile.cs [hunk 1]
```diff
@@ -18,6 +20,12 @@ public class SystemFile : IChildFile, IFastGetRoot
     /// <param name="path">The path to the file.</param>
     public SystemFile(string path)
     {
+        foreach (var c in global::System.IO.Path.GetInvalidPathChars())
+        {
+            if (path.Contains(c))
+                throw new FormatException($"Provided path contains invalid character: {c}");
```
Comment (d2dyno1) [inline src/System/IO/SystemFile.cs:26]:
```suggestion
                throw new FormatException($"Provided path contains invalid character '{c}'.");
```

File: src/System/IO/SystemFile.cs [hunk 2]
```diff
@@ -18,6 +20,12 @@ public class SystemFile : IChildFile, IFastGetRoot
...
+        }
+
         if (!File.Exists(path))
             throw new FileNotFoundException($"File not found at path {path}");
```
Comment (d2dyno1) [inline src/System/IO/SystemFile.cs:30]: Make sure to end exception messages with a dot (.)

File: src/Exceptions/FileAlreadyExistsException.cs [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+
+/// <summary>
+/// The exception that is thrown when attempting to create or recreate a file that already exists.
+/// </summary>
+public class FileAlreadyExistsException : IOException
+{
+    /// <summary>
+    /// Initializes a new instance of the <see cref="FileAlreadyExistsException"/> class.
+    /// </summary>
+    public FileAlreadyExistsException(string fileName) : base($"The file {fileName} already exists.")
```
Comment (d2dyno1) [inline src/Exceptions/FileAlreadyExistsException.cs:13]: You can also pass in HRESULT (0x80070050) = -2147024816 for the exception

File: src/System/IO/SystemFile.cs (hunk 2 above)
Comment (Arlodotexe) [reply src/System/IO/SystemFile.cs:30]: Good catch, I'll do a codebase sweep before we ship this.

File: src/Exceptions/FileAlreadyExistsException.cs (hunk 3 above)
Comment (Arlodotexe) [reply src/Exceptions/FileAlreadyExistsException.cs:13]: Nice, didn't see this. I'll give it a try.

Comment (Arlodotexe) [reply src/Exceptions/FileAlreadyExistsException.cs]: Implemented in 27ab5dd4f3b993891ca2380deb442103f48a9cc8, co-authored as well.

**Pull request merged**